jupyterlab_gcedetails/src/components/synchronized_sliders.tsx

```diff
@@ -0,0 +1,242 @@
+    return possibleValues.reduce((a, b) => {
+      return Math.abs(a - targetValue) < Math.abs(b - targetValue) ? a : b;
+    });
+  }
+
+  private onCpuChange(newCpu: number) {
+    this.setState({
+      cpuValue: newCpu,
+      cpuInputDisplayValue: String(newCpu),
+      memoryValue: this.memoryFromCpu(newCpu),
+      memoryInputDisplayValue: String(this.memoryFromCpu(newCpu)),
+    });
+
+    this.props.onCpuChange(newCpu);
+  }
+
+  private onMemoryChange(newMemory: number) {
+    this.setState({
+      cpuValue: this.cpuFromMemory(newMemory),
+      cpuInputDisplayValue: String(this.cpuFromMemory(newMemory)),
+      memoryValue: newMemory,
+      memoryInputDisplayValue: String(newMemory),
+    });
+
+    this.props.onMemoryChange(newMemory);
+  }
+
+  render() {
+    const {
+      cpuInputDisplayValue,
+      memoryInputDisplayValue,
+      cpuValue,
+      memoryValue,
+    } = this.state;
+
+    const handleCpuSliderChange = (event: any, newValue: number) => {
+      this.onCpuChange(newValue);
+    };
+
+    const handleMemorySliderChange = (event: any, newValue: number) => {
+      this.onMemoryChange(newValue);
+    };
+
+    const handleCpuInputChange = (
+      event: React.ChangeEvent<HTMLInputElement>
+    ) => {
+      this.setState({
+        cpuInputDisplayValue: event.target.value,
+      });
+    };
+
+    const handleCpuInputBlur = () => {
+      const newValue = this.closest(
+        Number(cpuInputDisplayValue),
+        this.cpuValues
+      );
+      this.onCpuChange(newValue);
+    };
+
+    const handleMemoryInputChange = (
+      event: React.ChangeEvent<HTMLInputElement>
+    ) => {
+      this.setState({
+        memoryInputDisplayValue: event.target.value,
+      });
+    };
+
+    const handleMemoryInputBlur = () => {
+      const newValue = this.closest(
+        Number(memoryInputDisplayValue),
+        this.memoryValues
+      );
+      this.onMemoryChange(newValue);
+    };
+
+    return (
+      <div>
+        <InputSlider
+          possibleValues={this.cpuValues}
+          label="Cores"
+          value={cpuValue}
+          inputDisplayValue={cpuInputDisplayValue}
+          handleSliderChange={handleCpuSliderChange}
+          handleInputChange={handleCpuInputChange}
+          handleInputBlur={handleCpuInputBlur}
+          helperText="1 - 96 vCPUs"
+        />
+        <InputSlider
+          possibleValues={this.memoryValues}
+          label="Memory"
+          value={memoryValue}
+          inputDisplayValue={memoryInputDisplayValue}
+          handleSliderChange={handleMemorySliderChange}
+          handleInputChange={handleMemoryInputChange}
+          handleInputBlur={handleMemoryInputBlur}
+          helperText="3.75 - 360 GB"
+        />
+      </div>
+    );
+  }
+}
+
+interface InputSliderProps {
+  value: number;
+  possibleValues: number[];
+  label?: string;
+  handleSliderChange: (event: any, newValue: number) => void;
+
```

Comment:
Weird that this space didn't get picked up by the linter

Reply:
Good catch!